Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -3782,33 +3802,114 @@ namespace AssetProcessor
                 {
                     continue;
                 }
 
                 for (const auto& thisEntry : resolvedDependencyList)
                 {
                     SourceFileDependencyEntry newDependencyEntry(
-                        builderId,
+                        builderId, entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(), thisEntry.toUtf8().constData(),
+                        SourceFileDependencyEntry::DEP_JobToJob, // significant line in this code block
+                        false);
+                    newDependencies.push_back(AZStd::move(newDependencyEntry));
+                }
+
+                // Source dependencies don't have any concept of jobs so if we store an entry for every job, we end up with duplicates.
+                // This isn't an issue with the builder, so no error/warning is needed, just check to avoid duplicates.
+                if (auto result = jobDependenciesDeduplication.emplace(builderId, entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(),
+                    resolvedDatabaseName.toUtf8().constData()); result.second)
+                {
+                    SourceFileDependencyEntry newDependencyEntry(
+                        builderId, entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(), resolvedDatabaseName.toUtf8().constData(),
+                        jobDependency.m_jobDependency.m_sourceFile.m_sourceDependencyType ==
+                                AssetBuilderSDK::SourceFileDependency::SourceFileDependencyType::Wildcards
+                            ? SourceFileDependencyEntry::DEP_SourceLikeMatch
+                            : SourceFileDependencyEntry::DEP_JobToJob, // significant line in this code block
+                        !entry.m_sourceFileInfo.m_uuid.IsNull());
+                    newDependencies.push_back(AZStd::move(newDependencyEntry));
+                }
+            }
+        }
+
+        AZStd::unordered_set<AZStd::string> resolvedSourceDependenciesDeduplication;
+
+        for (const AZStd::pair<AZ::Uuid, AssetBuilderSDK::SourceFileDependency>& sourceDependency : entry.m_sourceFileDependencies)
+        {
+            // figure out whether we can resolve the dependency or not:
+            QStringList resolvedDependencyList;
+            QString resolvedDatabaseName;
+            if (!ResolveSourceFileDependencyPath(sourceDependency.second, resolvedDatabaseName, resolvedDependencyList))
+            {
+                // ResolveDependencyPath should only fail in a data error, otherwise it always outputs something,
+                // even if that something starts with the placeholder.
+                continue;
+            }
+
+            // Handle multiple resolves (wildcard dependencies)
+            for (const auto& thisEntry : resolvedDependencyList)
+            {
+                if(jobDependenciesDeduplication.contains(DependencyDeduplication{sourceDependency.first, entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(),
```

Comment:
you may want to format this more vertically to make it more readable and match the other formats in this file